app/components/primer/conditional_wrapper.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module Primer
+  # Conditionally renders a `Primer::BaseComponent` around the given content. If the given condition
+  # is true, a `Primer::BaseComponent` will render around the content. If the condition is false, only
+  # the content is rendered.
+  #
+  # @example True conditional
+  #   <%# condition is true, so content will be wrapped in a <span> tag
+  #   <%= render Primer::ConditionalWrapper.new(condition: true, tag: :span, class: "foobar")) do %>
+  #     <%# also rendered %>
+  #     <p class="bazboo">Some text</p>
+  #   <% end %>
+  #
+  # @example False conditional
+  #   <%# condition is false so no <span> tag will render around the content (i.e. the <p> tag)
+  #   <%= render(Primer::ConditionalWrapper.new(condition: false, tag: :span, class: "foobar")) do %>
+  #     <%# this content will be rendered %>
+  #     <p class="bazboo">Some text</p>
+  #   <% end %>
+  #
+  # @param condition [Boolean] Whether or not to wrap the content in a `Primer::BaseComponent`.
+  # @param base_component_arguments [Hash] The arguments to pass to `Primer::BaseComponent`.
+  class ConditionalWrapper < Primer::Component
+    def initialize(condition:, **base_component_arguments)
+      @condition = condition
+      @base_component_arguments = base_component_arguments
+    end
+
+    def call
+      return content unless @condition
```

Comment:
Just mentioning since it seemed relevant, but it's likely that we'll end up evaluating `content` even if `@condition` is `false` if https://github.com/github/view_component/pull/1428 comes back (which I believe it will).

If we wanted to get ahead of that change, we could call `content` before the `return content` call above.

Reply:
Maybe ignore me, I was thinking this may have been similar to the Avo issue where there was a conditional component that would render `nothing`, relying on `content` never being evaluated. It looks like we always expect `content` to be used here. 🤦

Reply:
Right yeah, we always expect `content` to be evaluated/emitted.